RecoPixelVertexing/PixelTriplets/plugins/CAHitNtupletGeneratorKernelsImpl.h

```diff
@@ -561,11 +561,12 @@ __global__ void kernel_fillHitDetIndices(HitContainer const *__restrict__ tuples
 
 __global__ void kernel_fillNLayers(TkSoA *__restrict__ ptracks, cms::cuda::AtomicPairCounter *apc) {
   auto &tracks = *ptracks;
-  auto first = blockIdx.x * blockDim.x + threadIdx.x;
-  auto ntracks = apc->get().m;
+  uint32_t first = blockIdx.x * blockDim.x + threadIdx.x;
+  // clamp the number of tracks to the capacity of the SoA
+  auto ntracks = std::min<uint32_t>(apc->get().m, tracks.stride() - 1);
```

Comment:
@VinInn does this look like a good fix to you ?
An alternative could be to adapt all client code to check the size vs the stride (or look for the `nhits == 0` marker), but this seemed easier to implement and safer to use.

On the other this will completely silence the error.
Maybe we could check if `nTracks() == stride() - 1` and assume it to mean that an "overflow" occurred ?